### ISSUE TITLE
Failing tests on CI

### ISSUE THREAD

**Issue by teoljungberg** [calleerlandsson/pick#166]
I opened https://github.com/Homebrew/homebrew-core/pull/5608 to upgrade `pick` to v1.5.1, but the test suite seems to be constantly failing on CI. Homebrew's CI is on Jenkis, which I presume is running on a Mac.

Has this pattern been reproduced on Travis(or any other CI service)? With the hopes of preferably being fixed.

**Comment by mptre** [comment]
The cause of failure has been discussed in another PR[1].

[1] https://github.com/Homebrew/homebrew-core/pull/5079

**Comment by calleerlandsson** [comment]
Thanks for opening this issue, @teoljungberg, and thanks for linking to that PR, @mptre, I hadn't seen it before.

Yes, we're seeing similar failures on Travis CI in the test output for #165: https://travis-ci.org/calleerlandsson/pick/jobs/165200469

```
0.07s$ sh ./tests/test.sh ./tests/01.t
./tests/01.t: expected exit code 0, got 1
pick-test: /dev/tty: Device not configured
--- /var/folders/my/m6ynh3bn6tq06h7xr3js0z7r0000gn/T/pick.exp.XXXXXX.CbyHVkBa	2016-10-05 10:55:06.000000000 +0000
+++ /var/folders/my/m6ynh3bn6tq06h7xr3js0z7r0000gn/T/pick.act.XXXXXX.ibvkseHi	2016-10-05 10:55:06.000000000 +0000
@@ -1 +1 @@
-pick-test: /dev/tty: Device not configured
+1
```

**Comment by calleerlandsson** [comment]
I have released [v1.5.2](https://github.com/calleerlandsson/pick/releases/tag/v1.5.2) in the hopes of it fixing the CI issues on macOS.

**Comment by teoljungberg** [comment]
It doesn't seem to have been fixed, atleast according to the CI run on my PR to Homebrew.

Which can be found here: https://bot.brew.sh/job/Homebrew%20Core/9314/

**Comment by mptre** [comment]
Looks like it got a bit further than last time. At this point I would
guess pick(1) itself rather than pick-test(1) is exiting while trying to
open `/dev/tty`. Could you dump the contents of `test-suite.log` on
failure[1]?

[1] https://github.com/Homebrew/homebrew-core/pull/5079#issuecomment-249386864

**Comment by teoljungberg** [comment]
@mptre I don't know how I could dump the contents of that file without modifying the version of pick I'm building. Any suggestion on how to debug this?

**Comment by mptre** [comment]
On Thu, Oct 13, 2016 at 06:57:16AM -0700, Teo Ljungberg wrote:
> @mptre I don't know how I could dump the contents of that file without
> modifying the version of pick I'm building. Any suggestion on how to
> debug this?

Just change the formula[1].

[1] https://github.com/Homebrew/homebrew-core/pull/5079/commits/4b630ffb61144874836f8753740d916a4db3aeef

**Comment by teoljungberg** [comment]
@mptre Damn, that was easier than I thought.

Here's the output: https://gist.github.com/teoljungberg/f7610e0d38fc1a05452bccfe1f84f1b8

The problem seems to be isolated to:
```
TERM environment variable not set.
```

**Comment by mptre** [comment]
Interesting, what if you change the command to:

```sh
env TERM=xterm make check || (cat test-suite.log; exit 1)
```

**Comment by teoljungberg** [comment]
@mptre That did the trick.

I updated the PR to Homebrew to reflect that change. But, it might be good to have pick-tets(1) setting that environment variable if it doesn't exist.

**Comment by mptre** [comment]
Glad it worked out. I'm a bit sceptical to such a change since it seems
very tied to this environment. If the problem occurs in other
environments I might reconsider.

**Comment by teoljungberg** [comment]
My guess is that its a shared problem with all CI environments, if it is - we should probably have a generic solution for this. If it's only tied to Jenkins(the Homebrew CI), I agree with you.

**Comment by mptre** [comment]
Agree, it works fine on Travis at this point. This issue can be closed?
